lib/ansible/modules/cloud/amazon/iot_thing.py

```diff
@@ -0,0 +1,484 @@
+        update or delete the IOT thing will fail.
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+
+EXAMPLES = '''
+- name: create IOT thing
+  iot_thing:
+    name: mything
+    state: present
+
+- name: delete IOT thing
+  iot_thing:
+    name: mything
+    state: absent
+
+- name: set type of IOT thing
+  iot_thing:
+    name: mything
+    type: mytype
+
+- name: set the type associated with an IOT thing
+  iot_thing:
+    name: mything
+    remove_type: True
+
+- name: disassociate an IOT thing from a type
+  iot_thing:
+    name: mything
+    type:
+      state: absent
+
+- name: attach a certificate to an IOT thing
+  iot_thing:
+    name: mything
+    attached_principals:
+      - "arn:aws:iot:us-east-1:account-id:cert/uuid"
+
+- name: detach a certificate from an IOT thing
+  iot_thing:
+    name: mything
+    detached_principals:
+      - "arn:aws:iot:us-east-1:account-id:cert/uuid"
+
+- name: add attributes to an IOT thing
+  iot_thing:
+    name: mything
+    attributes:
+      key1: value1
+      key2: value2
+
+- name: remove attributes from an IOT thing
+  iot_thing:
+    name: mything
+    absent_attributes:
+      - key3
+'''
+
+
+RETURN = '''
+default_client_id:
+    description: The ID of the default client for this IOT thing.
+    returned: on success when I(state=present)
+    type: string
+    sample: mythingid
+version:
+    description: The current version of the thing record in the IOT registry.
+    returned: on success when I(state=present)
+    type: int
+    sample: 123
+attribute:
+    description: The attributes currently set on the IOT thing.
+    returned: on success when I(state=present)
+    type: dict
+    sample: {"attribute1": "value1", "attribute2", "value2"}
+principals:
+    description: List of certificate or credential ARNs currently attached to this IOT thing.
+    returned: on success when I(state=present)
+    type: list
+    sample: ["arn:aws:iot:us-east-1:account-id:cert/uuid-1", "arn:aws:iot:us-east-1:account-id:cert/uuid-2"]
+thing_arn:
+    description: The ARN of this IOT thing.
+    returned: on success when the IOT thing is initially created.
+    type: string
+    sample: "arn:aws:iot:us-east-1:account-id:cert/mything"
+thing_name:
+    description: The name of this IOT thing.
+    returned: on success when I(state=present)
+    type: string
+    sample: mything
+thing_type_name:
+    description: The name of this IOT thing.
+    returned: on success when I(state=present) and there is an IOT type associated with this IOT thing.
+    type: string
+    sample: mytype
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import get_aws_connection_info, boto3_conn, camel_dict_to_snake_dict
+
+
+try:
+    from botocore.exceptions import ClientError, ValidationError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
```

Comment:
All other modules have main at the bottom - having it here is a little unexpected